src/main/java/org/wso2/carbon/inbound/poll/dbeventlistener/DBEventPollingConsumer.java

```diff
@@ -0,0 +1,380 @@
+        }
+        return true;
+    }
+
+    private MessageContext createMessageContext() {
+        MessageContext msgCtx = synapseEnvironment.createMessageContext();
+        org.apache.axis2.context.MessageContext axis2MsgCtx = ((org.apache.synapse.core.axis2.Axis2MessageContext) msgCtx)
+                .getAxis2MessageContext();
+        axis2MsgCtx.setServerSide(true);
+        axis2MsgCtx.setMessageID(UUIDGenerator.getUUID());
+        msgCtx.setProperty(org.apache.axis2.context.MessageContext.CLIENT_API_NON_BLOCKING, true);
+        PrivilegedCarbonContext carbonContext = PrivilegedCarbonContext.getThreadLocalCarbonContext();
+        axis2MsgCtx.setProperty(MultitenantConstants.TENANT_DOMAIN, carbonContext.getTenantDomain());
+        return msgCtx;
+    }
+
+    /**
+     * Execute the query to retrieve the records, create each record as OMElement and inject to the sequence
+     */
+    private void fetchDataAndInject() {
+        Statement statement = null;
+        ResultSet rs = null;
+        String tableName = properties.getProperty(DBEventConstants.DB_TABLE);
+        DBEventRegistryHandler dbEventListnerRegistryHandler = new DBEventRegistryHandler();
+        String lastUpdatedTimestampFromRegistry = null;
+        if (filteringCriteria.equals(DBEventConstants.DB_FILTERING_BY_TIMESTAMP)) {
+            lastUpdatedTimestampFromRegistry = dbEventListnerRegistryHandler.readFromRegistry(registryPath).toString();
+        }
+        String dbScript = buildQuery(tableName, filteringCriteria,
+                filteringColumnName, lastUpdatedTimestampFromRegistry);
+        try {
+            if(!isConnectionAlive()) {
+                createConnection();
+            }
+            statement = connection.prepareStatement(dbScript);
+            rs = statement.executeQuery(dbScript);
+            ResultSetMetaData metaData = rs.getMetaData();
+            List<String> primaryKeys = Arrays.asList(primaryKeysFromConfig);
+            while (rs.next()) {
+                if (filteringCriteria.equals(DBEventConstants.DB_DELETE_AFTER_POLL)) {
+                    deleteQuery = "DELETE FROM " + tableName + " WHERE ";
+                } else if (filteringCriteria.equals(DBEventConstants.DB_FILTERING_BY_BOOLEAN)) {
+                    updateQuery = "UPDATE " + tableName + " SET " + filteringColumnName + "='false'" + " WHERE ";
+                }
+                OMFactory factory = OMAbstractFactory.getOMFactory();
+                OMElement result = factory.createOMElement("Record", null);
+                int count = metaData.getColumnCount();
+                for (int i = 1; i <= count; i++) {
+                    String columnName = metaData.getColumnName(i);
+                    int type = metaData.getColumnType(i);
+                    String columnValue = getColumnValue(rs, columnName, type);
+                    if (StringUtils.isNotEmpty(deleteQuery) && primaryKeys.contains(columnName)) {
```

Comment:
No need to validate this. Anyway user has given the primarykey. We should trust the parameter value given by user.